lib/ui/common/CircularProgress.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:flutter/material.dart';
+import 'dart:math';
+
+class CircularProgress extends StatefulWidget {
+  final Color _lineColor;
+  final double _progress;
+
+  const CircularProgress({Key key, @required double progress, Color lineColor})
+      : this._lineColor = lineColor ?? const Color(0xff24d900),
+        this._progress = progress,
```

Comment:
Add an assert here :D
```suggestion
        this._progress = progress,
        assert(progress >= 0 && progress <= 1, 'Progress should be between 0 and 1'),
```